### PR TITLE
Try MinimalH2AsyncClient (which is optimized for multiplexing)

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs please review and merge.

### CODE REVIEW

File: vespa-feed-client/src/main/java/ai/vespa/feed/client/ApacheCluster.java [hunk 1]
```diff
@@ -33,6 +33,13 @@
 class ApacheCluster implements Cluster {
 
     private final List<Endpoint> endpoints = new ArrayList<>();
+    private final List<BasicHeader> defaultHeaders = Arrays.asList(new BasicHeader("User-Agent", String.format("vespa-feed-client/%s", Vespa.VERSION)),
+                                                                   new BasicHeader("Vespa-Client-Version", Vespa.VERSION));
```
Comment (jonmv) [inline vespa-feed-client/src/main/java/ai/vespa/feed/client/ApacheCluster.java:37]: Do we need both of these?

Comment (jonmv) [reply vespa-feed-client/src/main/java/ai/vespa/feed/client/ApacheCluster.java:37]: I guess it's quite fast when compressed ...

Comment (jonmv) [reply vespa-feed-client/src/main/java/ai/vespa/feed/client/ApacheCluster.java:37]: (But encodeHeaders typically has most of the perf samples for small requests ...)

Comment (bjorncs) [reply vespa-feed-client/src/main/java/ai/vespa/feed/client/ApacheCluster.java:37]: Both are needed afaik. Former for access log, latter for metric.

**Review by bjorncs — APPROVED** [review]

**Review by bjorncs — APPROVED** [review]

**Pull request merged**